appi.adoc

```diff
@@ -0,0 +1,397 @@
+﻿[[appendix-examples-groups, Appendix I, Recommended Practices and Annotated Examples for Using Groups]]
+
+[appendix]
+== Recommended Practices and Annotated Examples for Using Groups
+
+This appendix contains recommended practices for using netCDF-4 groups in CF compliant data, as well as annotated examples of existing use cases for representing data using groups.
+
+=== Recommended practices
+
+. The use of attributes attached to a group in order to store metadata normally attached directly to variables is discouraged.
+This includes, for example, replacing per-variable attributes like `_FillValue`, `scale_factor`, `valid_min`, with group-level equivalents.
+Although group attributes might be more concise, it is likely to create problems with downstream software and to reduce interoperability.
+
+. Each group's attributes, dimensions and variables should be self-contained in combination with the group metadata and any inherited properties (e.g., dimension sizes, coordinates).
+This ensures that if a group is renamed or extracted (with any inherited properties) into a new file, its information content is preserved.
+For example, storage of ensembles as sibling groups (as described below) is often clearer when the realization number is encoded in the group name.
+In this case, the `Realization` group attribute retains the realization number even if the group is renamed.
+Other commonly enumerated group names, such as station identifiers, buoy numbers, or channel wavelengths are analogous.
+It is fine to enumerate or itemize names so long as the number or name is redundantly stored as a group attribute.
```

Comment:
I would _much_ rather say that a group attribute is the canonical way forward, but that you can redundantly use enumerated or itemized group names.

Reply:
What I meant to say is, I think: I would rather just drop the word "redundant" - the group attribute is metadata according to convention, after all.

Reply:
@davidhassell I agree. "redundant" is not the best wording. I suggest we replace it with "also".

Reply:
I'll leave it as-is unless someone has strong feelings for the purpose of minimising changes at this stage.

Reply:
Oops, sorry for the noise - I thought I was commenting elsewhere. I've adopted the wording in 9e5fd72.